### PR TITLE
test: run test suite on macos on master

### PR DESCRIPTION
**Pull request by XVincentX**
Run harness test suite on macOS when we merge something in master and when we're releasing stuff. Closes #442 

SO-149

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,28 +57,29 @@ jobs:
...
+    executor: << parameters.executor >>
     steps:
       - checkout
       - run:
           name: Install Dependencies
           command: yarn
       - run:
           name: Install dockerize
-          command: wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && sudo tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && rm dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz
+          command: curl -OL https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-$(uname)-amd64-$DOCKERIZE_VERSION.tar.gz && sudo tar -C /usr/local/bin -xzvf dockerize-$(uname)-amd64-$DOCKERIZE_VERSION.tar.gz
```
Comment (P0lip) [inline .circleci/config.yml]: Keep in mind `uname` returns `Linux` than `linux` when executed under Linux. I guess this doesn't matter, though.

Comment (P0lip) [inline .circleci/config.yml]: I'd download it to `/opt` or even better to `/tmp`, unpack there and move afterwards.

File: .circleci/config.yml [hunk 2]
```diff
@@ -116,9 +123,18 @@ workflows:
...
+          name: harness-macos
+          executor: macos
           filters:
             tags:
               only: /.*/
+      - harness:
+          executor: docker-node
+          filters:
+            branches:
+              only: master
```
Comment (P0lip) [inline .circleci/config.yml]: I guess you might want to include develop here later on.

Comment (XVincentX) [reply .circleci/config.yml]: I'll do whatever I can to *not* use git flow in Prism. It's solving problems that Prism does not have.

File: .circleci/config.yml (hunk 1 above)
Comment (XVincentX) [reply .circleci/config.yml]: Apparently works, so I'll not touch it :trollface:

**Pull request merged**